### PR TITLE
rclip: init at 1.7.6

### PR DESCRIPTION
**Pull request by iynaix**
## Description of changes

Adds [rclip](https://github.com/yurijmikhalevich/rclip), an AI-Powered Command-Line Photo Search Tool.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/open-clip-torch/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    torchvision
+    tqdm
+  ];
+
+  pythonImportsCheck = [ "open_clip" ];
+
+  meta = with lib; {
+    description = "An open source implementation of CLIP";
+    homepage = "https://github.com/mlfoundations/open_clip";
+    license = licenses.mit;
```
Comment (natsukium) [inline pkgs/development/python-modules/open-clip-torch/default.nix:51]:
```suggestion
    license = licenses.asl20;
```
https://github.com/mlfoundations/open_clip/blob/29b90b8dde9125c479c0ddbab477310f665a0435/setup.py#L39

File: pkgs/development/python-modules/open-clip-torch/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    huggingface-hub
+    protobuf
+    regex
+    sentencepiece
+    timm
+    torch
+    torchvision
+    tqdm
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/open-clip-torch/default.nix:45]: Could you please enable tests?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

File: pkgs/by-name/rc/rclip/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    open-clip-torch
+    pillow
+    requests
+    torch
+    torchvision
+    tqdm
+  ];
+
+  pythonImportsCheck = [ "rclip" ];
+
```
Comment (natsukium) [inline pkgs/by-name/rc/rclip/package.nix:31]: ditto

**Comment by iynaix** [comment]
Disabled new failing tests for open-clip-torch this time.

File: pkgs/development/python-modules/open-clip-torch/default.nix [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    pandas
+    transformers
+    webdataset
+  ];
+
+  pythonImportsCheck = [ "open_clip" ];
+
+  pytestFlagsArray = lib.optionals (stdenv.isAarch64 || stdenv.isDarwin) [
+    "--ignore=tests/test_wds.py"
+  ];
```
Comment (natsukium) [inline pkgs/development/python-modules/open-clip-torch/default.nix:63]:
```suggestion
  disabledTestPaths = lib.optionals (stdenv.isAarch64 || stdenv.isDarwin) [
    "tests/test_wds.py"
  ];
```

File: pkgs/by-name/rc/rclip/package.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  ];
+
+  nativeCheckInputs = with python3Packages; [ pytestCheckHook ];
+
+  pythonImportsCheck = [ "rclip" ];
+
+  pytestFlagsArray = [
+    # requires network
+    "--ignore=tests/e2e/test_rclip.py"
+  ];
```
Comment (natsukium) [inline pkgs/by-name/rc/rclip/package.nix:37]:
```suggestion
  disabledTestPaths = [
    # requires network
    "tests/e2e/test_rclip.py"
  ];
```

**Review by natsukium — COMMENTED** [review]
Besides this, LGTM.

File: pkgs/development/python-modules/open-clip-torch/default.nix (hunk 4 above)
Comment (iynaix) [reply pkgs/development/python-modules/open-clip-torch/default.nix]: Ok, made the changes.

File: pkgs/by-name/rc/rclip/package.nix (hunk 5 above)
Comment (iynaix) [reply pkgs/by-name/rc/rclip/package.nix]: Ok, made the changes.

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**